### ISSUE TITLE
Merge in branch `goexperiment-rangefunc` when range-over-func is released

### ISSUE THREAD

**Issue by jbduncan** [jbduncan/go-containers#8]
Blocked on the currently-experimental [range-over-func](https://github.com/golang/go/issues/61405) Go feature being merged into a stable version of Go.

Merge in branch https://github.com/jbduncan/go-containers/tree/goexperiment-rangefunc, which introduces and uses range-over-func method(s).

**Comment by jbduncan** [comment]
This was done a while ago, so this can be closed.
